src/optimagic/optimizers/iminuit_migrad.py

```diff
@@ -33,19 +32,30 @@
 @dataclass(frozen=True)
 class IminuitMigrad(Algorithm):
     stopping_maxfun: int = STOPPING_MAXFUN
-    stopping_maxiter: int = STOPPING_MAXITER
-    errordef: Optional[float] = None
 
     def _solve_internal_problem(
         self, problem: InternalOptimizationProblem, params: NDArray[np.float64]
     ) -> InternalOptimizeResult:
         def wrapped_objective(x: NDArray[np.float64]) -> float:
             return float(problem.fun(x))
 
-        m = Minuit(
-            wrapped_objective,
-            params,
-        )
+        wrapped_gradient = None
+        jac_func = None
+        if problem.jac is not None:
+
+            def wrapped_gradient(x: NDArray[np.float64]) -> NDArray[np.float64]:
+                return problem.jac(x)
+
+            jac_func = problem.jac
+        elif problem.fun_and_jac is not None:
+
+            def wrapped_gradient(x: NDArray[np.float64]) -> NDArray[np.float64]:
+                _, jac = problem.fun_and_jac(x)
+                return jac
```

Comment:
Again, `fun_and_jac` is never None for the `InternalOptimizationProblem`. The logic you have here (including much more actually) is already in the `InternalOptimizationProblem` so we don't have to repeat it per optimizer.